extensions/amp-form/0.1/amp-form.js

```diff
@@ -652,6 +694,33 @@ export class AmpForm {
     return this.waitOnPromisesOrTimeout_(varSubPromises, 100);
   }
 
+  /**
+   * Call getValue() on Async Input elements, and
+   * Create hidden inputs containing their returned values
+   * @param {!Element} asyncInput
+   * @return {!Promise}
+   * @private
+   */
+  getValueForAsyncInput_(asyncInput) {
+    return asyncInput.getImpl().then(implementation => {
+      return implementation.getValue().then(value => {
```

Comment:
nit:`Promise` chains should almost always be simplified to be one-level deep.

```js
return asyncInput.getImpl()
    .then(implementation => implementation.getValue())
    .then(value => {
      /* ... */
    });
```
 This avoids the Promise Pyramid of Doom. 

> ["You can easily smell a promise pyramid whenever you see then() blocks going inwards in an existing then() block."](https://web.archive.org/web/20180301090527/http://blog.ministryofprogramming.com/prevent-pyramids-with-flat-promise-chains/)